liquid-rust-typeck/src/checker.rs

```diff
@@ -302,6 +305,33 @@ impl<'a, 'tcx> Checker<'a, 'tcx> {
         Ok(())
     }
 
+    fn check_assert(
+        &mut self,
+        env: &mut TypeEnv<'tcx>,
+        cursor: &mut Cursor,
+        cond: &Operand,
+        expected: &bool,
+        target: BasicBlock,
+    ) -> Result<(), ErrorReported> {
+        let cond_ty = self.check_operand(env, cond);
+
+        let pred = match cond_ty.kind() {
+            TyKind::Refine(BaseTy::Bool, e) => {
+                e.clone()
+            }
+            _ => unreachable!("unexpected cond_ty {:?}", cond_ty),
+        };
+
+        let assert = ExprKind::BinaryOp(BinOp::Eq, pred, ExprKind::Constant(liquid_rust_fixpoint::Constant::Bool(*expected)).intern()).intern();
```

Comment:
I would go with the following as I think it'll generate a slightly simpler constraint:

```suggestion
        let assert = if cond { pred } else { pred.not() }
```